Bottom: 3736927f2f621ae83d4830280ac3a3ab93274c06
Top:    0c7d9ceab0d5df15acf756ee788c788d99747f7d
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-14 20:57:46 +0800

vhost_net: honor batched dequeuing for busy polling

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index e3d7ea1..b51a7df 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -601,10 +601,13 @@ static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 	return len;
 }
 
-static int sk_has_rx_data(struct sock *sk)
+static int sk_has_rx_data(struct vhost_net_virtqueue *vq, struct sock *sk)
 {
 	struct socket *sock = sk->sk_socket;
 
+	if (vq->rx_array)
+		return vhost_net_buf_peek(vq);
+
 	if (sock->ops->peek_len)
 		return sock->ops->peek_len(sock);
